test/sinon-test.js

```diff
@@ -504,6 +504,18 @@
                 };
 
                 assert(sinon.deepEqual(obj1, obj2));
+            },
+
+            "passes objects without a prototype": function () {
+                function Obj() {}
+                Obj.prototype = Object.create(null);
+
+                var obj = new Obj();
```

Comment:
Simplify the test a bit:

```js
var obj = Object.create(null);
obj.foo = "bar";
```

Also there a [few variations of this test](https://github.com/sinonjs/sinon/blob/master/test/util/core/deep-equal-test.js#L266-L307) in master. Could you cover all these bases please?